### PR TITLE
[Feature 7] testing add follow use case testing

### PR DESCRIPTION
**Pull request by jannater**
- Added Integration tests
- Added Unit tests
- Refactored FollowController to RestController
- Refactored FollowUseCase to check already following

### CODE REVIEW

File: src/test/java/ca/utoronto/fitbook/integration/FollowControllerIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    public void followUserExpectsOkResponse() throws Exception {
+        Map<String, String> payload = new HashMap<>();
+        payload.put("followerId", this.testUser1.getId());
+        payload.put("followeeId", this.testUser2.getId());
+        ObjectMapper mapper = new ObjectMapper();
+
+        this.mockMvc.perform(post("/follow").session(authorizedSession).content(mapper.writeValueAsString(payload))
+                        .contentType(MediaType.APPLICATION_JSON_VALUE))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON));
```
Comment (youssefsoli) [inline src/test/java/ca/utoronto/fitbook/integration/FollowControllerIntegrationTest.java:97]: You should check that the repository has the correct following lists saved.

File: src/test/java/ca/utoronto/fitbook/unit/FollowUseCaseTest.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    }
+
+
+    @Test
+    public void followUserReturnsValidResponse() {
+        FollowCommand followCommand = new FollowCommand(testUser1.getId(), testUser2.getId());
+
+        FollowResponse followResponse = this.followUseCase.followUser(followCommand);
+
+        Assertions.assertEquals(testUser2.getId(), followResponse.getFolloweeId());
```
Comment (youssefsoli) [inline src/test/java/ca/utoronto/fitbook/unit/FollowUseCaseTest.java:77]: You should check that the repository was modified correctly.

File: src/test/java/ca/utoronto/fitbook/unit/FollowUseCaseTest.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+        Assertions.assertThrows(FollowService.UserSelfFollowingException.class, ()->this.followUseCase.followUser(followCommand));
+    }
+
+    @Test
+    public void followUserRepeatsFollowThrowsUserAlreadyFollowingException() {
+        testUser1.setFollowingIdList(List.of(testUser2.getId()));
+        testUser2.setFollowersIdList(List.of(testUser1.getId()));
+        userLocalMemoryRepository.save(testUser1);
+        userLocalMemoryRepository.save(testUser2);
```
Comment (youssefsoli) [inline src/test/java/ca/utoronto/fitbook/unit/FollowUseCaseTest.java:91]: Modification and saving of the user should not be happening inside of tests if those users are being used by other tests, instead, make new users that are following each other.

**Review by youssefsoli — CHANGES_REQUESTED** [review]
Looks good, just some things that should be verified in the test cases to make sure that we actually persist to the database.

**Review by youssefsoli — APPROVED** [review]
Everything looks good after my comments were applied, good to merge.

**Review by OrbitalVitamin — APPROVED** [review]
Testing and changes are appropriate. Seems like all possible test cases are covered in this change.

**Review by jpablo2002 — APPROVED** [review]
Great changes addressing the issue of following a user multiple times, as well as general testing!

**Pull request merged**